src/components/application_manager/src/hmi_capabilities_impl.cc

```diff
@@ -1319,29 +1543,409 @@ bool HMICapabilitiesImpl::load_capabilities_from_file() {
-      set_vehicle_type(vehicle_type_so);
+
+    if (!json_root_getter.GetInterfaceJsonMember(hmi_interface::vehicle_info)
+             .isNull()) {
+      auto vehicle_info_main_node =
+          json_root_getter.GetMainJsonMember(hmi_interface::vehicle_info);
+      auto vehicle_info_override_node =
+          json_root_getter.GetOverrideJsonMember(hmi_interface::vehicle_info);
+      JsonCapabilitiesGetter json_vehicle_info_getter(
+          vehicle_info_main_node, vehicle_info_override_node);
+      auto vehicle_type_node = json_vehicle_info_getter.GetJsonMember(
+          hmi_response::vehicle_type,
+          hmi_apis::FunctionID::VehicleInfo_GetVehicleType,
+          interfaces_to_update_);
+      if (!vehicle_type_node.isNull()) {
+        smart_objects::SmartObject vehicle_type_so;
+        formatters::CFormatterJsonBase::jsonValueToObj(vehicle_type_node,
+                                                       vehicle_type_so);
+        set_vehicle_type(vehicle_type_so);
+      }
     }  // VehicleType end
   } catch (...) {
     return false;
   }
   return true;
 }
 
+hmi_apis::Common_Language::eType
+HMICapabilitiesImpl::GetActiveLanguageForInterface(
+    const char* interface_name) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (strcmp(hmi_interface::ui, interface_name) == 0) {
+    return active_ui_language();
+  }
+  if (strcmp(hmi_interface::vr, interface_name) == 0) {
+    return active_vr_language();
+  }
+  if (strcmp(hmi_interface::tts, interface_name) == 0) {
+    return active_tts_language();
+  }
+  return hmi_apis::Common_Language::INVALID_ENUM;
+}
+
+std::vector<hmi_apis::FunctionID::eType>
+HMICapabilitiesImpl::GetInterfacesToUpdate() const {
+  return interfaces_to_update_;
+}
+
+bool HMICapabilitiesImpl::AreAllFieldsSaved(
+    const Json::Value& root_node,
+    const char* interface_name,
+    const std::vector<std::string>& sections_to_check) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (!check_existing_json_member(root_node, interface_name)) {
+    LOG4CXX_DEBUG(logger_,
+                  interface_name
+                      << " interface is not found. All fields should be saved");
+    return false;
+  }
+
+  const auto& interface_node = root_node.get(interface_name, Json::Value::null);
+  for (auto it = sections_to_check.begin(); it != sections_to_check.end();
+       ++it) {
+    const std::string section = (*it).c_str();
+    if (!check_existing_json_member(interface_node, section.c_str())) {
+      LOG4CXX_DEBUG(logger_,
+                    "Field " << *it << " should be saved into the file");
+      return false;
+    }
+
+    if (hmi_response::language == section) {
+      const auto active_language =
+          GetActiveLanguageForInterface(interface_name);
+      const auto json_language = interface_node[hmi_response::language];
+
+      if (active_language !=
+          MessageHelper::CommonLanguageFromString(json_language.asString())) {
```

Comment:
@LitvinenkoIra I suggest to create an `std::string` variable and use the variable for the more readable condition. For example:
```
auto json_language_str = MessageHelper::CommonLanguageFromString(json_language.asString();
if(active_language ! json_language_str)
...
```

Reply:
@IGapchuk not sure about this, the function itself is long, don't wanna add extra lines